src/bg/user-script-registry.js

```diff
@@ -290,6 +290,14 @@ async function saveUserScript(userScript) {
       onSaveError(e.target.error);
       reject(e);
       db.close();
+    } finally {
+      // Send the script change event, even though the save may have failed.
```

Comment:
This worries me.  If the save failed, I don't want the editor changing, and maybe throwing away unsaved details.

Reply:
The `userScript` that's being saved should represent the editor as it is. You're basically returning the same details you submitted to the background, along with any downloaded resources. You shouldn't lose any unsaved details as `userScript` represents all those unsaved details (that failed to save).

Reply:
Hm, I suppose that if the external resources take a long time to download time you may be able to perform some edits before the async save is returned. But that issue applies to both successful saves and unsuccessful saves.

I think the best thing to do is popup some modal that says 'saving' and doesn't go away until the result is back. If a save error occurs then display the error in the modal. If no errors, then auto close.